c2rust-analyze/src/context.rs

```diff
@@ -577,6 +600,43 @@ where
     })
 }
 
+/// Renumber the keys of `ptr_info`, producing a new table.  For a new `PointerId` `q`, the
+/// `PointerInfo` in the output is computed by merging the values of `ptr_info[p]` for all `p`
+/// where `map[p] == q`.
+fn remap_global_ptr_info(
+    ptr_info: &GlobalPointerTable<PointerInfo>,
+    map: &GlobalPointerTable<PointerId>,
+    num_pointers: usize,
+) -> GlobalPointerTable<PointerInfo> {
+    let mut new_info = GlobalPointerTable::<PointerInfo>::new(num_pointers);
+    for (old, &new) in map.iter() {
+        new_info[new] |= ptr_info[old];
+    }
+    new_info
+}
+
+/// Renumber the keys of `old_local_ptr_info`, producing a new local table.  `new_global_ptr_info`
+/// must already be remapped with `remap_global_ptr_info`.
```

Comment:
nit: is it worth wrapping `new_global_ptr_info` in a type so that the invariant becomes type safe? seems easy to mess this up otherwise by calling before `remap_global_ptr_info` is called

Reply:
Do you mean the `` `new_global_ptr_info` must already be remapped with `remap_global_ptr_info` `` invariant?

Reply:
yes

Reply:
I think it's not worth it because this function is called in only one place in the entire codebase.  But I'll think about it a little and see if there's an easy refactor to make it more robust.